### DISCUSSION TITLE
cloaking and forwarding don't seem to work on my setup

### DISCUSSION THREAD

**Discussion by dev8765** [DNSCrypt/dnscrypt-proxy#2075] [Potential issues]
./dnscrypt-proxy -version
2.1.0

./dnscrypt-proxy -check
root@openwrt:~/dnscryptproxy/linux-arm# ./dnscrypt-proxy -check
[2022-03-20 10:58:04] [NOTICE] Configuration successfully checked

./dnscrypt-proxy -resolve example.com
Resolving [example.com] using 127.0.0.1 port 54
Resolver      : 74.63.24.239
Canonical name: example.com.
IPv4 addresses: 93.184.216.34
IPv6 addresses: 2606:2800:220:1:248:1893:25c8:1946
Name servers  : b.iana-servers.net., a.iana-servers.net.
DNSSEC signed : yes
Mail servers  : 1 mail servers found
HTTPS alias   : -
HTTPS info    : -
Host info     : -
TXT records   : wgyf8z8cgvm2qmxpnbnldrcltvk4xqfn, v=spf1 -all

dnscrypt-proxy doesn't resolve "localhost" and "openwrt.lan" (at least with the default configuration)
1. I tried "cloaking-rules.txt"
2. I tried "forwarding-rules.txt" with "lan 192.168.1.1" uncommented
I had both files uncommented in "dnscrypt-proxy.toml":
[NOTICE] Loading the set of cloaking rules from [cloaking-rules.txt]
But to no avail, "localhost" and "openwrt.lan" keep unresolved

**Comment by jedisct1** [comment]
What do the following commands return?

```sh
dnscrypt-proxy -resolve localhost,192.168.1.1

dnscrypt-proxy -resolve openwrt.lan,192.168.1.1
```

?

**Comment by dev8765** [comment]
My production instance of dnscrypt-proxy is listening on port 53, with no cloaking/forwarding configured: 

`dnscrypt-proxy -resolve localhost,192.168.1.1
Resolving [localhost] using 192.168.1.1 port 53
Resolver      : 74.63.24.237
Lying         : no
DNSSEC        : yes, the resolver supports DNSSEC
Canonical name: localhost.
IPv4 addresses: -
IPv6 addresses: -
Name servers  : name does not exist
DNSSEC signed : no
Mail servers  : no mail servers found
HTTPS alias   : -
HTTPS info    : -
Host info     : -
TXT records   : -`

`dnscrypt-proxy -resolve openwrt.lan,192.168.1.1
Resolving [openwrt.lan] using 192.168.1.1 port 53
Resolver      : 74.63.24.237
Lying         : no
DNSSEC        : yes, the resolver supports DNSSEC
Canonical name: openwrt.lan.
IPv4 addresses: -
IPv6 addresses: -
Name servers  : name does not exist
DNSSEC signed : no
Mail servers  : no mail servers found
HTTPS alias   : -
HTTPS info    : -
Host info     : -
TXT records   : -`

My test instance is listening on port 54, with cloaking/forwarding configured:

`./dnscrypt-proxy -resolve localhost
Resolving [localhost] using 127.0.0.1 port 54
Resolver      : 45.63.110.187 (draco.plan9-ns2.com.)
Canonical name: localhost.
IPv4 addresses: 127.0.0.1
IPv6 addresses: ::1
Name servers  : name does not exist
DNSSEC signed : no
Mail servers  : no mail servers found
HTTPS alias   : -
HTTPS info    : -
Host info     : -
TXT records   : -`

`./dnscrypt-proxy -resolve openwrt.lan
Resolving [openwrt.lan] using 127.0.0.1 port 54
Resolver      : 45.63.110.187 (draco.plan9-ns2.com.)
Canonical name: openwrt.lan.
IPv4 addresses: -
IPv6 addresses: -
Name servers  : name does not exist
DNSSEC signed : no
Mail servers  : no mail servers found
HTTPS alias   : -
HTTPS info    : -
Host info     : -
TXT records   : -`

**Reply by jedisct1** [reply]
Looks like `192.168.1.1` doesn't respond to DNS queries, at least not for `.lan`.

**Comment by dev8765** [comment]
cloaking-rules.txt:
Sorry my fault, I had a typo
"localhost":      It's fully working 
"openwrt.lan":  It's mostly working: 
nslookup openwrt.lan
Name:      openwrt.lan
Address 1: 192.168.1.1
*** Can't find openwrt.lan: No answer

forwarding-rules.txt:
Without an entry for "openwrt.lan" in "forwarding-rules.txt" and with the line
lan             192.168.1.1
uncommented, resolving of "openwrt.lan" is not working:
nslookup openwrt.lan
# Waiting ca. 10 seconds 
;; connection timed out; no servers could be reached
